numpy/core/src/umath/ufunc_object.c

```diff
@@ -575,12 +770,42 @@ get_ufunc_arguments(PyUFuncObject *ufunc,
     int nout = ufunc->nout;
     int nop = ufunc->nargs;
     PyObject *obj, *context;
-    PyObject *str_key_obj = NULL;
     const char *ufunc_name = ufunc_get_name_cstr(ufunc);
-    int has_sig = 0;
-
     /*
-     * Initialize objects so caller knows when outputs and other optional
+     * Keyword definitions.
+     *
+     * Note: if any changes are made, one also needs to change
+     * the initalization of the outputs in the kwds loop.
+     */
+    static PyObject **kwnames[] = {
```

Comment:
I wonder if I should instead do
```
PyObject *kwnames[13] = {NULL};
```
and then explicitly fill on first call, to get rid of the extra indirections every else.

Reply:
Only if profiling shows it's worth it. I'd leave that to a future PR.